src/components/common/ColonyFunding/ColonyFunding.tsx

```diff
@@ -34,112 +34,103 @@ const MSG = defineMessages({
 
 const ColonyFunding = () => {
   const { formatMessage } = useIntl();
+  const isMobile = useMobile();
+  const { colony } = useColonyContext();
+  const { domains, tokens } = colony || {};
 
   const [selectedDomainId, setSelectedDomainId] = useState<number>(
     COLONY_TOTAL_BALANCE_DOMAIN_ID,
   );
 
-  // const domainChoices = useMemo<
-  //   ComponentProps<typeof Select>['options']
-  // >(() => {
-  //   if (!data || !data.processedColony) {
-  //     return [];
-  //   }
-  //   const {
-  //     processedColony: { domains },
-  //   } = data;
-  //   return [
-  //     {
-  //       value: COLONY_TOTAL_BALANCE_DOMAIN_ID.toString(),
-  //       label: { id: 'domain.all' },
-  //     },
-  //     ...sortBy(
-  //       domains.map(({ ethDomainId, name }) => ({
-  //         label: name,
-  //         value: ethDomainId.toString(),
-  //       })),
-  //       ['value'],
-  //     ),
-  //   ];
-  // }, [data]);
+  const domainChoices = useMemo<
+    ComponentProps<typeof Select>['options']
+  >(() => {
+    if (!domains?.items) {
+      return [];
+    }
+    return [
+      {
+        value: COLONY_TOTAL_BALANCE_DOMAIN_ID.toString(),
+        label: { id: 'domain.all' },
+      },
+      ...(domains?.items ?? [])
+        .map((domain) => ({
+          label: domain?.name ?? '',
+          value: domain?.nativeId.toString() ?? '',
+        }))
+        .sort(
+          (first, second) =>
+            parseInt(first.value, 10) - parseInt(second.value, 10),
+        ),
+    ];
+  }, [domains]);
 
-  const selectedDomainLabel = 'Dummy Domain Label';
-  // const selectedDomainLabel: string = useMemo(() => {
-  //   const { label = '' } =
-  //     domainChoices.find(
-  //       ({ value }) => value === selectedDomainId.toString(),
-  //     ) || {};
-  //   return typeof label === 'string' ? label : formatMessage(label);
-  // }, [domainChoices, formatMessage, selectedDomainId]);
+  const selectedDomainLabel: string = useMemo(() => {
+    const { label = '' } =
+      domainChoices.find(
+        ({ value }) => value === selectedDomainId.toString(),
+      ) || {};
+    return typeof label === 'string' ? label : formatMessage(label);
+  }, [domainChoices, formatMessage, selectedDomainId]);
 
-  // if (
-  //   loading ||
-  //   (data?.colonyAddress &&
-  //     !data.processedColony &&
-  //     !((data.colonyAddress as any) instanceof Error))
-  // ) {
-  //   return (
-  //     <div className={styles.loadingWrapper}>
-  //       <LoadingTemplate loadingText={MSG.loadingText} />
-  //     </div>
-  //   );
-  // }
-
-  // if (!colonyName || error || !data?.processedColony) {
-  //   console.error(error);
-  //   return <NotFoundRoute />;
-  // }
-
-  // const { processedColony: colony } = data;
+  // eslint-disable-next-line react/no-unstable-nested-components
+  const Aside = () => (
```

Comment:
While I appreciate this, and agree with you that is should live outside, we're going to change the whole UI in like 2-3 months. 

I'm not going to waste time improving **Dapp** component logic _(which this is, since every component I've added was copy pasted from there)_, only to have to remove it in the very near future